Makefile

```diff
@@ -1,40 +1,72 @@
+python-dir=./python/src
+node-dir=./node/src
+
 # Anything that needs to be done before the other rules run
 setup:
 	mkdir -p build/data
 
 # Runs the entire pipeline using real data sources
 release: setup
 	curl 'https://data.smgov.net/resource/w8ue-6cnd.csv?$$limit=50000' \
-	  | node parse-trees.js \
-	  | python pruning_planting.py \
-	  | node download-images.js \
-	  | node split-trees.js build/data
+	  | node $(node-dir)/parse-trees.js \
+	  | python $(python-dir)/pruning_planting.py \
+	  | node $(node-dir)/download-images.js \
+	  | node $(node-dir)/split-trees.js build/data
+
+release-docker: setup
+	curl 'https://data.smgov.net/resource/w8ue-6cnd.csv?$$limit=50000' \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data public-tree-map-node:latest node /public-tree-map/parse-trees.js \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data public-tree-map-python:latest python /public-tree-map/pruning_planting.py \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data public-tree-map-node:latest node /public-tree-map/download-images.js \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data -v $$(pwd)/build:/public-tree-map/build public-tree-map-node:latest node /public-tree-map/split-trees.js build/data
 
 # Runs the pipeline using local data, but skips the CPU-intensive python tasks
 img-test: setup
 	cat data/trees.csv \
-	  | node parse-trees.js \
-	  | node download-images.js \
-	  | node split-trees.js build/data
+	  | node $(node-dir)/parse-trees.js \
+	  | node $(node-dir)/download-images.js \
+	  | node $(node-dir)/split-trees.js build/data
+
+img-test-docker: setup
+	cat data/trees.csv \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data public-tree-map-node:latest node /public-tree-map/parse-trees.js \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data public-tree-map-node:latest node /public-tree-map/download-images.js \
+	  | docker run -i --rm -v $$(pwd)/data:/public-tree-map/data -v $$(pwd)/build:/public-tree-map/build public-tree-map-node:latest node /public-tree-map/split-trees.js build/data
 
 # Runs the pipeline, but skips downloading images
 no-images: setup
 	curl 'https://data.smgov.net/resource/w8ue-6cnd.csv?$$limit=50000' \
-	  | node parse-trees.js \
-	  | node split-trees.js build/data
+	  | node $(node-dir)/parse-trees.js \
+	  | node $(node-dir)/split-trees.js build/data
+
+no-images-docker: setup
+	curl 'https://data.smgov.net/resource/w8ue-6cnd.csv?$$limit=50000' \
```

Comment:
This won't work, proper link is 'https://data.smgov.net/resource/w8ue-6cnd.csv?$limit=5000'

Reply:
See comment above